src/cpp/core/include/core/http/TcpIpAsyncClientSsl.hpp

```diff
@@ -62,6 +69,28 @@ class TcpIpAsyncClientSsl
             if (ec)
                LOG_ERROR(Error(ec, ERROR_LOCATION));
          }
+
+      #ifdef _WIN32
+         // on Windows, OpenSSL does not support loading certificates from the Windows certificate store
+         // because of this, each time we need to verify certificates, we initialize
+         // all certificates individually with OpenSSL
+         const WindowsCertificateStore& certStore = getCertificateStore();
+         for (const auto& cert : certStore.certificates)
+         {
+            if (X509_STORE* store = SSL_CTX_get_cert_store(sslContext_.native_handle()))
+            {
+               if (::X509_STORE_add_cert(store, cert) != 1)
+               {
+                  boost::system::error_code ec = rstudio_boost::system::error_code(
+                              static_cast<int>(::ERR_get_error()),
+                              boost::asio::error::get_ssl_category());
+                  Error error(ec, ERROR_LOCATION);
+                  error.addProperty("Description", "Could not add Windows certificate to OpenSSL cert store");
```

Comment:
If there are a bunch of certificates in the Windows store and a couple fail to be added to the OpenSSL cert store, will the logs indicate which ones? (looks like they won't unless it's in the error object somewhere that I missed)

Reply:
No. This would be difficult to do correctly (would need a lot of dev time + testing to figure out how to extract the CA name to add to the log information, was not easy for me to figure out looking at openssl documentation), which indeed makes this log error less useful than I wanted it to be. However, I figured leaving it in would be worth something, as it could possibly help us troubleshoot an install in the wild that isn't validating properly.

Reply:
Actually I'll look into this - found an article talking about this.

Reply:
Okay, I was hoping it would be more straightforward!